netsim/roles/bridge.py

```diff
@@ -0,0 +1,110 @@
+  link_found = False
+
+  for link in topology.get('links',[]):
+    if 'vlan' in link:                                      # A link already has VLAN parameters, skip it
+      continue
+    for intf in link.get('interfaces',[]):
+      if 'vlan' in intf:                                    # The interface has VLAN parameters, skip it
+        continue
+      n_role = topology.nodes[intf.node].get('role','router')
+      if n_role != 'bridge':                                # Not a bridge interface, skip it
+        continue
+      intf.vlan.access = BR_DEFAULT
+      link_found = True
+      append_to_list(link,'_br_list',intf)                  # Remember the bridges we found
+
+  return link_found
+
+"""
+When exactly one of the nodes on link with more than two nodes is a bridge,
+expand the link into multiple P2P links with the bridge node. Print a warning
+(and do nothing) if there are more than two bridges attached to the link
+"""
+def expand_multiaccess_links(topology: Box) -> None:
+  for link in list(topology.get('links',[])):
+    if '_br_list' not in link:                              # Is this one of the relevant links?
+      continue                                              # No, move on
+
+    br_list = link._br_list                                 # Get the list of bridges
+    link.pop('_br_list',None)                               # ... and remove it from the link
+
+    if not br_list:                                         # Empty list?
+      continue                                              # Weird, but we've seen weirder things
+
+    if len(br_list) > 1:                                    # Multiple bridges on multi-access links confuse us
+      br_names = [ intf.node for intf in br_list ]
+      log.warning(
+        text=f'Multiple bridges ({",".join(br_names)}) attached to multi-access link {link._linkname}',
+        more_hints='netlab can expand only multi-access links attached to a single bridge',
+        module='bridge')
+      continue
+
+    br_intf = br_list[0]                                    # Remember the bridge interface
+    link_cnt = 0
+    for intf in link.interfaces:
+      if intf != br_intf:                                   # Did we find an interesting interface?
+        l_data = get_box(link)                              # Copy the link data
+        link_cnt += 1
+        l_data._linkname = f'{link._linkname}.{link_cnt}'   # Create unique link and and linkindex
+        l_data.linkindex = links.get_next_linkindex(topology)
+        l_data.interfaces = [ get_box(br_intf), intf ]      # ... recreate P2P interfaces
+        topology.links.append(l_data)                       # ... and append the new P2P link to the links
+
+    topology.links.remove(link)                             # Finally, remove original link
+
+def pre_transform(topology: Box) -> None:
+  if not create_default_VLAN(topology):                     # Add default bridge VLAN
```

Comment:
Most devices I'm familiar with - Dell OS10, EOS, FRR - would have an (implicit) default VLAN also in 'router' mode.

Perhaps default VLAN creation would fit better in the VLAN module?

Reply:
> Most devices I'm familiar with - Dell OS10, EOS, FRR - would have an (implicit) default VLAN also in 'router' mode.

Layer-3 switches have a default VLAN.

> Perhaps default VLAN creation would fit better in the VLAN module?

This is not adding the "default VLAN" functionality to the **vlan** module. This is an implementation of a very simple bridge using the **vlan** module functionality, and as that one needs **vlans**, we have to use them.